terraform/modules/shared/main.tf

```diff
@@ -472,8 +472,49 @@ resource "azurerm_kubernetes_cluster" "cluster" {
   }
 
   identity {
-    type         = "UserAssigned"
-    identity_ids = [azurerm_user_assigned_identity.pipeline_runner.id]
+    type = "SystemAssigned"
+  }
+
+}
+
+data "azurerm_kubernetes_cluster" "credentials" {
+  name                = azurerm_kubernetes_cluster.cluster.name
+  resource_group_name = var.resource_group_name
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.azurerm_kubernetes_cluster.credentials.kube_config.0.host
+    client_certificate     = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.client_certificate)
+    client_key             = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.client_key)
+    cluster_ca_certificate = base64decode(data.azurerm_kubernetes_cluster.credentials.kube_config.0.cluster_ca_certificate)
+  }
+}
+
+resource "helm_release" "record_linkage" {
+  name          = "phdi-${terraform.workspace}"
+  repository    = "https://cdcgov.github.io/phdi-charts/"
+  chart         = "record-linkage-chart"
+  recreate_pods = true
+
+  set {
+    name  = "image.tag"
+    value = "latest"
+  }
+
+  set {
+    name  = "databasePassword"
+    value = azurerm_postgresql_flexible_server.mpi.administrator_password
+  }
+
+  set {
+    name  = "databaseName"
+    value = azurerm_postgresql_flexible_server_database.mpi.name
+  }
+
+  set {
+    name  = "databaseHost"
```

Comment:
A potential optimization down the line would be to reference these as part of the Key Vault instead of grabbing the direct values through Terraform. However, this is how we're doing it for the existing container applications, and it's seemingly not any less secure than storing the values in KV.

Reply:
I think this is fine. In some ways this is a little more secure, because when values are in a KV users can go in and read them via the portal or CLI. If we can keep all the creds handled internal to terraform that seems like a good way to go.